### PR TITLE
[Fixe 270] Create github action to triage issues automatically

### LINKED ISSUE #270: Set up GitHub Action to triage issues automatically when an issue has a status
## Problem

Currently, we need to triage our issues [in the backlog](https://github.com/centerofci/mathesar/projects/2) manually.

## Proposed solution

We should set up [this action](https://github.com/marketplace/actions/move-labeled-or-milestoned-issue-to-a-specific-project-column) (or something similar) with the following rules:
- `type: meta` issues should be in the `Meta Issues` column of the Backlog project, regardless of status label.
- `status: ready` issues should be in the `Ready` column of the Backlog project
- `status: detail needed` issues should be in the `Detail Needed` column of the Backlog project
- `status: blocked` issues should be in the `Blocked` column of the Backlog project
- `status: future` issues should be removed from the Backlog project and put in the Future project (no specific column is necessary)

## Additional context
This wiki page might be helpful: [Issue Triage](https://wiki.mathesar.org/en/engineering/issue-triage)

### PR DESCRIPTION
**Pull request by linzhou-zhong**
Fixes #270 

**Technical details**

**Screenshots**

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .github/workflows/put-labeled-issue-in-backlog-column.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      - name: Move meta issue ticket to the Meta column
+        uses: konradpabjan/move-labeled-or-milestoned-issue@v2.0
+        with:
+          project-url: "https://github.com/centerofci/mathesar/projects/2"
+          label-name: "type: meta"
+          column-name: "Meta issues"
+          columns-to-ignore: "Meta issues"
+          action-token: "${{ secrets.MATHESAR_GITHUB_TOKEN }}"
+
+      - name: Move meta ready ticket to the Ready column
```
Comment (kgodey) [inline .github/workflows/put-labeled-issue-in-backlog-column.yml:20]: Can you remove the `meta` from this line (and all other lines except the "meta issues" step

File: .github/workflows/put-labeled-issue-in-backlog-column.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      uses: konradpabjan/move-labeled-or-milestoned-issue@v2.0
+      with:
+        project-url: "https://github.com/centerofci/mathesar/projects/2"
+        label-name: "status: ready"
+        column-name: "Ready"
+        columns-to-ignore: "Meta issues"
```
Comment (kgodey) [inline .github/workflows/put-labeled-issue-in-backlog-column.yml:26]: Do we also need to ignore "Ready" here?

File: .github/workflows/put-labeled-issue-in-backlog-column.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        action-token: "${{ secrets.MATHESAR_GITHUB_TOKEN }}"
+
+    - name: Move meta detail ticket to the Detail needed column
+      uses: konradpabjan/move-labeled-or-milestoned-issue@v2.0
+      with:
+        project-url: "https://github.com/centerofci/mathesar/projects/2"
+        label-name: "status: detail needed"
+        column-name: "Detail needed"
+        columns-to-ignore: "Detail needed"
```
Comment (kgodey) [inline .github/workflows/put-labeled-issue-in-backlog-column.yml:35]: Should we also ignore the Meta issues column here?

File: .github/workflows/put-labeled-issue-in-backlog-column.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        action-token: "${{ secrets.MATHESAR_GITHUB_TOKEN }}"
+
+    - name: Move meta detail ticket to the Detail needed column
+      uses: konradpabjan/move-labeled-or-milestoned-issue@v2.0
+      with:
+        project-url: "https://github.com/centerofci/mathesar/projects/2"
+        label-name: "status: blocked"
+        column-name: "Blocked"
+        columns-to-ignore: "Blocked"
```
Comment (kgodey) [inline .github/workflows/put-labeled-issue-in-backlog-column.yml:44]: Should we also ignore the Meta issues column here?

**Review by kgodey — CHANGES_REQUESTED** [review]
@linzhou-zhong Looks great! Thanks for creating your own action to support this.

Added a couple of questions and a minor comment.

File: .github/workflows/put-labeled-issue-in-backlog-column.yml (hunk 2 above)
Comment (linzhou-zhong) [reply .github/workflows/put-labeled-issue-in-backlog-column.yml:26]: Of course, I'll add Ready in `columns-to-ignore` so we don't need a duplicated check.

File: .github/workflows/put-labeled-issue-in-backlog-column.yml (hunk 3 above)
Comment (linzhou-zhong) [reply .github/workflows/put-labeled-issue-in-backlog-column.yml]: Added

File: .github/workflows/put-labeled-issue-in-backlog-column.yml (hunk 4 above)
Comment (linzhou-zhong) [reply .github/workflows/put-labeled-issue-in-backlog-column.yml]: Added

Comment (linzhou-zhong) [reply .github/workflows/put-labeled-issue-in-backlog-column.yml]: Added

File: .github/workflows/put-labeled-issue-in-backlog-column.yml (hunk 1 above)
Comment (linzhou-zhong) [reply .github/workflows/put-labeled-issue-in-backlog-column.yml]: Removed

**Review by kgodey — APPROVED** [review]
Looks good. I made some more changes for accuracy in the text (and fixed a typo) in https://github.com/centerofci/mathesar/pull/622/commits/04bad0b55a1fbe76f3f06be9aaf4fbe1636689fe.

**Pull request merged**